crates/common/src/msgs/data_requests/sudo/mod.rs

```diff
@@ -1,24 +1,68 @@
+use crate::types::{Bytes, U128};
+
 pub mod expire_data_requests;
 pub mod remove_requests;
 
 #[cfg_attr(feature = "cosmwasm", cosmwasm_schema::cw_serde)]
 #[cfg_attr(not(feature = "cosmwasm"), derive(serde::Serialize, Debug, PartialEq))]
 #[cfg_attr(not(feature = "cosmwasm"), serde(rename_all = "snake_case"))]
-pub struct RemoveDataRequest {
-    pub dr_id: String,
+pub struct DistributionBurn {
+    /// The amount to burn
+    pub amount: U128,
 }
 
-impl From<RemoveDataRequest> for crate::msgs::SudoMsg {
-    fn from(value: RemoveDataRequest) -> Self {
-        SudoMsg::RemoveDataRequest(value).into()
-    }
+#[cfg_attr(feature = "cosmwasm", cosmwasm_schema::cw_serde)]
+#[cfg_attr(not(feature = "cosmwasm"), derive(serde::Serialize, Debug, PartialEq))]
+#[cfg_attr(not(feature = "cosmwasm"), serde(rename_all = "snake_case"))]
+pub struct DistributionSend {
+    /// The address to send the funds to.
+    pub to:     Bytes,
+    /// The amount to send to the address.
+    pub amount: U128,
+}
+
+#[cfg_attr(feature = "cosmwasm", cosmwasm_schema::cw_serde)]
+#[cfg_attr(not(feature = "cosmwasm"), derive(serde::Serialize, Debug, PartialEq))]
+#[cfg_attr(not(feature = "cosmwasm"), serde(rename_all = "snake_case"))]
+pub enum DistributionKind {
+    /// For burning funds
+    Burn(DistributionBurn),
+    /// For sending the funds to someone
+    Send(DistributionSend),
+}
+
+#[cfg_attr(feature = "cosmwasm", cosmwasm_schema::cw_serde)]
+#[cfg_attr(not(feature = "cosmwasm"), derive(serde::Serialize, Debug, PartialEq))]
+#[cfg_attr(not(feature = "cosmwasm"), serde(rename_all = "snake_case"))]
+pub enum DistributionType {
+    TallyReward,
+    ExecutorReward,
+    RelayerReward,
+    TimedOut,
```

Comment:
We haven't decided what happens with the funds when a timeout happens, nice one!

Reply:
Not for this repo, but we discussed that we could have a tally parameter for a fixed gas amount for timeout DRs. That would mean that timeout DRs will have a small cost and the rest can be refunded to the requestor.